server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/ServerApiHelper.java

```diff
@@ -107,9 +107,12 @@ public CompletableFuture<HttpClient.Response> rawGetAsync(String relativePath) {
   }
 
   private String buildEndpointUrl(String relativePath) {
+    return concat(endpointParams.getBaseUrl(), relativePath);
+  }
+
+  public static String concat(String baseUrl, String relativePath) {
     var fullUrl = new StringBuilder();
-    var endpointUrl = endpointParams.getBaseUrl();
-    fullUrl.append(endpointUrl.endsWith("/") ? endpointUrl.substring(0, endpointUrl.length() - 1) : endpointUrl);
+    fullUrl.append(baseUrl.endsWith("/") ? baseUrl.substring(0, baseUrl.length() - 1) : baseUrl);
```

Comment:
StringUtils.removeEnd() ?

Reply:
I simplified the code. To me the real move would be to not use a String to represent a URL, and it would help with those operations